rdt/hyper_transformer.py

```diff
@@ -155,18 +194,28 @@ def fit(self, data):
             data (pandas.DataFrame):
                 Data to fit the transformers to.
         """
-        if self.transformers is not None:
-            self._transformers = load_transformers(self.transformers)
-        else:
-            self._transformers = self._analyze(data)
+        self._input_columns = list(data.columns)
+        self._update_field_types(data)
+        fitted_fields = set()
+
+        # Loop through field_transformers that are first level
+        for field in self.field_transformers:
+            if self._field_in_data(field, data):
+                if self._field_in_set(field, fitted_fields):
+                    warnings.warn(FIELD_ALREADY_FIT_WARNING)
```

Comment:
As mentioned above, I think this should raise a `ValueError`
Also, I would try to do this validation in the `__init__`, since at that time we already know all the values given by the user.